util/include/DigestImpl.ipp

```diff
@@ -22,31 +22,47 @@
 
 namespace concord::util::digest {
 
-class DigestUtil {
+class DigestCreator {
  public:
+  virtual ~DigestCreator() = default;
+
+  virtual void init() = 0;
+  virtual void update(const char* data, size_t len) = 0;
+  virtual void finish(char* outDigest) = 0;
+};
+
+// Implements digest creator using Crypto++ library.
+class CryptoppDigestCreator : public DigestCreator {
+ public:
+  CryptoppDigestCreator();
   static size_t digestLength();
   static bool compute(const char* input, size_t inputLength, char* outBufferForDigest, size_t lengthOfBufferForDigest);
+  void init() override {}
+  void update(const char* data, size_t len) override;
+  void finish(char* outDigest) override {}
+  void writeDigest(char* outDigest);
+  virtual ~CryptoppDigestCreator();
 
-  class Context {
-   public:
-    Context();
-    void update(const char* data, size_t len);
-    void writeDigest(char* outDigest);  // write digest to outDigest, and invalidate the Context object
-    ~Context();
-
-   private:
-    void* internalState;
-  };
+ private:
+  void* internalState;
```

Comment:
rename to internalState_

Reply:
Addressed https://github.com/arc-vmware/concord-bft/pull/3/commits/93e565e22eae96ba5c40756c974a391f8bcb282d